lisa/base_tools/service.py

```diff
@@ -200,3 +209,93 @@ def _check_service_running(self, name: str) -> bool:
 
 def _check_error_codes(cmd_result: ExecutableResult, error_code: int = 0) -> None:
     cmd_result.assert_exit_code(expected_exit_code=[0, error_code])
+
+
+class WindowsServiceStatus(int, Enum):
+    CONTINUE_PENDING = 5
+    PAUSE_PENDING = 6
+    PAUSED = 7
+    RUNNING = 4
+    START_PENDING = 2
+    STOP_PENDING = 3
+    STOPPED = 1
+    NOT_FOUND = 0
+
+
+class WindowsService(Tool):
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    @property
+    def command(self) -> str:
+        return ""
+
+    def enable_service(self, name: str) -> None:
+        pass
+
+    def restart_service(self, name: str, ignore_exit_code: int = 0) -> None:
+        self.node.tools[PowerShell].run_cmdlet(
+            f"Restart-service {name}",
+            force_run=True,
+        )
+        self.wait_for_service_start(name)
+
+    def stop_service(self, name: str) -> None:
+        self.node.tools[PowerShell].run_cmdlet(
+            f"Stop-Service {name} -Force",
+            force_run=True,
+            output_json=True,
+            fail_on_error=False,
+        )
+        self.wait_for_service_stop(name)
+
+    def wait_for_service_start(self, name: str) -> None:
+        self._wait_for_service(name, WindowsServiceStatus.RUNNING)
+
+    def wait_for_service_stop(self, name: str) -> None:
+        self._wait_for_service(name, WindowsServiceStatus.STOPPED)
+
+    def _check_exists(self) -> bool:
+        return True
+
+    def check_service_exists(self, name: str) -> bool:
+        if (
+            self._get_status(name, fail_on_error=False)
+            == WindowsServiceStatus.NOT_FOUND
+        ):
+            return False
+        return True
+
+    def _get_status(
+        self, name: str = "", fail_on_error: bool = True
+    ) -> WindowsServiceStatus:
+        try:
+            service_status = self.node.tools[PowerShell].run_cmdlet(
+                f"Get-Service {name}",
+                force_run=True,
+                output_json=True,
+            )
+            return WindowsServiceStatus(int(service_status["Status"]))
```

Comment:
Move this line to the end of the method, so it's easy to read.

Reply:
Fixed as suggested